### PR TITLE
done some adjustment to improve the button atom

### PR DESCRIPTION
**Pull request by ahmedelsayad11**
-modified the name structure
- added some needed values to the tailwind config 
- added other variants and sizes for the button to make it more reusable
- adjusted the types to match the new values 
- renamed the SVG icon to be more descriptive

### CODE REVIEW

**Review by Nilegfx — CHANGES_REQUESTED** [review]
can you please elaborate on the motive behind this PR?
this change enables any props to be passed to the Button component which is kind of anti-pattern. why would you do so and not being more specific about the reusable props you pass?

**Comment by ahmedelsayad11** [comment]
we faced some problem when trying to use this atom later, some button in the design does not match the atom (button) in Figma, 
[like the button of the post creation in the mobile view, it does have different padding than all other button and the cancel button too, have a different background and text color  so on ]
so I had to make it more flexible so we can reuse it again, besides the style we face another problem with the type of the button
it was made to be [type='button '] but happened to use it as a [type='submit'] and when I tried to make it accept a prop as the value for the [type], husky gave me an error, that I should write a static type for the button, so the way to make it work as we want to make whoever is gonna use it have the ability to overwrite the type of the button.
- to sum up these changes' purpose is to make the button more reusable.

**Comment by Nilegfx** [comment]
@elhaw please check this PR when you have some time.

File: modules/shared/components/atoms/Button/Button.tsx [hunk 1]
```diff
@@ -14,6 +14,7 @@ const Button: FC<IButton.IProps> = ({
   rightIcon = false,
   onlyIcon = false,
   onClick,
+  ...props
```
Comment (elhaw) [inline modules/shared/components/atoms/Button/Button.tsx:17]: we need to remove the `...props` and the prop that we need.

File: modules/shared/components/atoms/Button/types/IButton.d.ts [hunk 2]
```diff
@@ -2,7 +2,8 @@ import type * as EButton from './EButton';
 
 declare namespace IButton {
   export interface IProps extends IHandlers {
-    children?: string;
```
Comment (elhaw) [inline modules/shared/components/atoms/Button/types/IButton.d.ts:5]: we need to add the type of the new prop.

**Review by elhaw — CHANGES_REQUESTED** [review]
- The naming conventions for this component `ButtonArrowDown` misleading. It indicates to be a button but it is an image

File: modules/shared/components/atoms/Button/Button.tsx [hunk 3]
```diff
@@ -36,27 +40,28 @@ const Button: FC<IButton.IProps> = ({
...
+      /* eslint-disable react/button-has-type */
+      type={buttonType}
       onClick={onClick}
       className={btnClasses}
       disabled={disabled}
     >
       {leftIcon && (
         <span>
-          <ButtonArrowDown />
+          <ArowDown />
```
Comment (elhaw) [inline modules/shared/components/atoms/Button/Button.tsx:51]: I know it was my suggestion 😂😂 to name it `<ArrowDown />` but the more explicit the better. 
`<ArrowDownIcon />` is better. or you can suggest a better one.

File: modules/shared/components/atoms/Button/types/EButton.ts [hunk 4]
```diff
@@ -1,10 +1,13 @@
 export enum buttonVariantValues {
   PRIMARY = 'primary',
   SECONDARY = 'secondary',
+  Tertiary = 'tertiary',
   TEXT = 'text',
```
Comment (elhaw) [inline modules/shared/components/atoms/Button/types/EButton.ts:5]: to follow the same naming conventions we need to name it `quaternary `

File: modules/shared/components/atoms/Button/types/IButton.d.ts [hunk 5]
```diff
@@ -2,13 +2,15 @@ import type * as EButton from './EButton';
...
+    [property: string]: unknown;
+    children?: string | null;
     disabled?: boolean;
     variant: EButton.buttonVariantValues;
     size: EButton.buttonSizeValues;
     leftIcon?: boolean;
     rightIcon?: boolean;
     onlyIcon?: boolean;
+    buttonType: 'button' | 'reset' | 'submit' | undefined;
```
Comment (elhaw) [inline modules/shared/components/atoms/Button/types/IButton.d.ts:13]: we can omit the `undefined` type value

**Review by elhaw — CHANGES_REQUESTED** [review]

**Review by elhaw — CHANGES_REQUESTED** [review]
I found this erorr in storybook 

> (51:10) /Users/abdulrahman/mystories/pickify-frontend/modules/shared/components/atoms/Button/Button.module.css The `px-xll` class does not exist, but `px-xl` does. If you're sure that `px-xll` exists, make sure that any `@import` statements are being properly processed before Tailwind CSS sees your CSS, as `@apply` can only be used for classes in the same CSS tree.

  49 | /* the only of it is kind  -the responsive cancel button-  */
  50 | .btn-normal.btn-quaternary {
> 51 |   @apply px-xll py-1.5;
     |          ^
  52 | }
  53 |

**Review by elhaw — APPROVED** [review]
LGTM

**Review by elhaw — APPROVED** [review]
@Nilegfx  
your  comment on this PR is  : 

> this change enables any props to be passed to the Button component which is a kind of anti-pattern. why would you do so and not being more specific about the reusable props you pass?

- we added the Required prop  `ButtonType : 'submit' | 'reset' | 'button' ` 
I will merge the PR  as you may be busy checking the changes mentioned a bove

**Comment by elhaw** [comment]
@Nilegfx  
your  comment on this PR is  : 

> this change enables any props to be passed to the Button component which is a kind of anti-pattern. why would you do so and not being more specific about the reusable props you pass?

- we added the Required prop  `ButtonType : 'submit' | 'reset' | 'button' ` 
I will merge the PR  as you may be busy checking the changes mentioned above
if you found any issue regarding it, please comment and we will consider it.
